internal/profile/profile.go

```diff
@@ -0,0 +1,265 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package profile
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+
+	// DefaultProfile is the name of the default profile
+	DefaultProfile = "default"
+)
+
+// ErrNotAProfile is returned in cases where we don't have a valid profile directory
+var ErrNotAProfile = errors.New("Is is not a profile")
+
+// ConfigFile is a type for for the config file names in a managed profile config
+type ConfigFile string
+
+// ManagedProfileFiles is the list of all files managed in a profile
+// If you create a new file that's managed by a profile, it needs to go in this list
+var ManagedProfileFiles = map[ConfigFile]NewConfig{
+	KibanaConfigFile:              NewKibanaConfig,
+	PackageRegistryDockerfileFile: NewPackageRegistryDockerfile,
+	PackageRegistryConfigFile:     NewPackageRegistryConfig,
+	SnapshotFile:                  NewSnapshotFile,
+	PackageProfileMetaFile:        CreateProfileMetadata,
+	KibanaHealthCheckFile:         NewKibanaHealthCheck,
+}
+
+// ConfigProfile manages a stack config
```

Comment:
Shouldn't say stack config anymore as we've decided that profiles could be used for configuration beyond just stack-related configuration.

Reply:
Yah, good point. There's probably a few of those hanging around.